trin-core/src/portalnet/protocol.rs

```diff
@@ -215,11 +215,17 @@ impl PortalnetProtocol {
             .await
             .map_err(|e| e.to_string())?;
 
+        let overlay = Arc::new(Overlay::new(
```

Comment:
Everything in `Overlay` is in an `Arc` already, so you could probably `derive(Clone)` on it, and save a level of indirection.

Reply:
To clarify, you are suggesting that we `derive(Clone)` for `Overlay`? If so, wouldn't we still require `Arc<Overlay>` to share it between threads? I think there's some piece that I'm missing here.

Reply:
Yep, that's exactly what I'm suggesting.

Instead of passing an `Arc<Overlay>` between threads, you can just clone a single `Overlay` and share it between threads. The `Arc`s inside will still point to the same data.